pkg/controller/pod.go

```diff
@@ -18,26 +18,56 @@ import (
 	"encoding/json"
 
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 var podSpecFields = []string{"jobTemplate", "spec", "template"}
 
-// GetPodSpec looks inside arbitrary YAML for a PodSpec
-func GetPodSpec(obj map[string]interface{}) (*corev1.PodSpec, error) {
+// GetPodMetadataAndSpec looks inside arbitrary YAML for a PodSpec and it's metadata
+func GetPodMetadataAndSpec(obj map[string]any) (*metav1.ObjectMeta, *corev1.PodSpec, error) {
```

Comment:
I would get the metadata in another method, however, as I assume this is called a lot, that would double the processing time
So I took advantage and implemented along the spec search, this way, it should not affect performance that much

Reply:
Smart! Yeah this looks good